docs/developer-guide/nodejs-api.md

```diff
@@ -144,6 +146,22 @@ console.log(code.text);     // "var foo = bar;"
 
 In this way, you can retrieve the text and AST used for the last run of `linter.verify()`.
 
+## linter
+
+This API is an instance of `Linter` class as defined [above](#Linter). This is API is deprecated (as of 4.0.0) and will be removed in future release.
```

Comment:
This makes it slightly unclear what is deprecated, because people might not realize that `eslint.linter` is a separate object. I think we should clearly indicate what was deprecated and what it should be replaced with, e.g.

```markdown
The `eslint.linter` object (deprecated) is an instance of the `Linter` class as defined [above](#Linter).

`eslint.linter` exists for backwards compatibility, but we do not recommend using it because any mutations to it are shared among every module that uses `eslint`. Instead, please create your own instance of `eslint.Linter`.
```